tests-e2e/membership_transitions_test.go

```diff
@@ -397,6 +398,140 @@ func TestInviteRejectionTwice(t *testing.T) {
 	})
 }
 
+func TestLeavingRoomReturnsOneEvent(t *testing.T) {
+	alice := registerNewUser(t)
+	bob := registerNewUser(t)
+	roomName := "It's-a-me-invitio"
+
+	for _, aliceSyncing := range []bool{false, true} {
+		t.Run(fmt.Sprintf("leaving a room returns one leave event (multiple poller=%v)", aliceSyncing), func(t *testing.T) {
+			inviteRoomID := alice.CreateRoom(t, map[string]interface{}{"preset": "private_chat", "name": roomName})
+			t.Logf("TestLeavingRoomReturnsOneEvent room %s", inviteRoomID)
+
+			if aliceSyncing {
+				alice.SlidingSync(t, sync3.Request{})
+			}
+
+			// sync as bob, we see no invites yet.
+			res := bob.SlidingSync(t, sync3.Request{
+				Lists: map[string]sync3.RequestList{
+					"a": {
+						Ranges: sync3.SliceRanges{{0, 20}},
+						Filters: &sync3.RequestFilters{
+							IsInvite: &boolTrue,
+						},
+					},
+				},
+			})
+			m.MatchResponse(t, res, m.MatchList("a", m.MatchV3Count(0)), m.MatchRoomSubscriptionsStrict(nil))
+
+			// now invite bob
+			alice.InviteRoom(t, inviteRoomID, bob.UserID)
+			// sync as bob until we see the room
+			res = bob.SlidingSyncUntilMembership(t, res.Pos, inviteRoomID, bob, "invite")
+			t.Logf("bob is invited")
+
+			// join the room
+			bob.JoinRoom(t, inviteRoomID, []string{})
+			res = bob.SlidingSyncUntilMembership(t, res.Pos, inviteRoomID, bob, "join")
+			t.Logf("bob joined")
+
+			// leave the room again, we should receive exactly one leave response
+			bob.LeaveRoom(t, inviteRoomID)
+			res = bob.SlidingSyncUntilMembership(t, res.Pos, inviteRoomID, bob, "leave")
+
+			if room, ok := res.Rooms[inviteRoomID]; ok {
+				// If alice is NOT syncing, we run into this failure mode
```

Comment:
Can you confirm what part of this test was failing without the diff to `user.go`?

Reply:
If Alice is _NOT_ syncing, we receive the leave event twice, once through  `Accumulate` and once through `OnLeftRoom`. 

If Alice _IS_ syncing however, the test fails for exactly the same reason - we receive the same event twice, but at a different place in the test. We're syncing as Bob until we left the room, after that we do another `SlidingSync` call, which again receives the event.